packages/editor/src/components/post-publish-panel/index.js

```diff
@@ -153,12 +148,8 @@ export default compose( [
 		const postType = getPostType( getEditedPostAttribute( 'type' ) );
 
 		return {
-			hasPublishAction: get(
-				getCurrentPost(),
-				[ '_links', 'wp:action-publish' ],
-				false
-			),
-			isPostTypeViewable: get( postType, [ 'viewable' ], false ),
+			hasPublishAction: getCurrentPost()._links?.[ 'wp:action-publish' ],
```

Comment:
`getCurrentPost()` will always be an object:

https://github.com/WordPress/gutenberg/blob/4944315cac156c78313121657dff9d58ce8f4d3f/packages/editor/src/store/selectors.js#L170